lib/icingadb/icingadb-objects.cpp

```diff
@@ -1101,9 +1109,34 @@ void IcingaDB::UpdateState(const Checkable::Ptr& checkable)
 
 	Dictionary::Ptr stateAttrs = SerializeState(checkable);
 
-	m_Rcon->FireAndForgetQuery({"HSET", m_PrefixConfigObject + objectType + ":state", objectKey, JsonEncode(stateAttrs)}, Prio::RuntimeStateSync);
-	m_Rcon->FireAndForgetQuery({"HSET", m_PrefixConfigCheckSum + objectType + ":state", objectKey, JsonEncode(new Dictionary({{"checksum", HashValue(stateAttrs)}}))}, Prio::RuntimeStateSync);
+	String redisStateKey = m_PrefixConfigObject + objectType + ":state";
+	String redisChecksumKey = m_PrefixConfigCheckSum + objectType + ":state";
+	String checksum = HashValue(stateAttrs);
+
+	if (mode == StateUpdate::Full || mode == StateUpdate::Volatile) {
+		m_Rcon->FireAndForgetQueries({
+			{"HSET", redisStateKey, objectKey, JsonEncode(stateAttrs)},
+			{"HSET", redisChecksumKey, objectKey, JsonEncode(new Dictionary({{"checksum", checksum}}))},
+		}, Prio::RuntimeStateSync);
+	}
+
+	if (mode == StateUpdate::Full || mode == StateUpdate::RuntimeOnly) {
```

Comment:
I clearly prefer a switch (w/o default). But I don't demand it. Remember this the next time when you review a PR of mine.

Reply:
Which I still think is a bad idea as explained in https://github.com/Icinga/icinga2/pull/9146#discussion_r780138457 (and my later reply there)

Reply:
And I consider your idea bad. But I tolerate it.